### PR TITLE
add `SimplifySearchResults ` to hide the nodes without any matched parent or child from the search results

### PR DESCRIPTION
**Pull request by huangjia2107**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  add `SimplifySearchResults` to hide the nodes without any matched parent or child from the search results  |
| 🇨🇳 Chinese |  添加 `SimplifySearchResults` 来简化搜索结果的显示，隐藏父节点或子节点均无匹配的节点  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/tree/style/mixin.less [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import '../../style/mixins/index';
+﻿@import '../../style/mixins/index';
```
Comment (ElderJames) [inline components/tree/style/mixin.less:1]: Please fix the encoding.

File: components/tree/style/mixin.less
Comment (ElderJames) [inline components/tree/style/mixin.less:1]: Please rename the file with 'patch.less'.

File: components/tree/style/mixin.less (hunk 1 above)
Comment (huangjia2107) [reply components/tree/style/mixin.less:1]: The file encoding is changed from utf-8 to utf-8 with bom，so the changing chars is �ｿ

File: components/tree/style/mixin.less
Comment (huangjia2107) [reply components/tree/style/mixin.less:1]: fixed in the latest commit

**Review by ElderJames — APPROVED** [review]

**Pull request merged**